lib/IRGen/MetadataRequest.cpp

```diff
@@ -1102,9 +1102,6 @@ MetadataAccessStrategy irgen::getTypeMetadataAccessStrategy(CanType type) {
       assert(type->hasUnboundGenericType());
     }
 
-    if (type->isForeignReferenceType())
-      return MetadataAccessStrategy::PublicUniqueAccessor;
-
     if (requiresForeignTypeMetadata(nominal))
       return MetadataAccessStrategy::ForeignAccessor;
```

Comment:
This is what needs to be returned for a C++ reference type.

Reply:
I see.  We were lazily emitting the type metadata when we first referenced in, but we were emitting it with strong linkage?  Yes, this is the right fix, then.

Reply:
Yes, that's right